Fix random memMemoryAllocator errors

Wanted9811 opened the pull request.

Wanted9811 committed:
Fix random memMemoryAllocator errors

Wanted9811 committed:
Styling fix

Wanted9811 committed:
Set old size of vehSiren to 0

0x1F9F1 commented on src/handlers/feature_handlers.cpp:
```diff
@@ -6628,8 +6628,8 @@ void vehSirenHandler::Reset() {
 
-void vehSirenHandler::SizeOf() {
-    hook::StaticThunk<0x577360>::Call<vehSiren*>(0x164);
+void vehSirenHandler::OperatorNew() {
```
You are replacing a call to a function with the type `void * __cdecl operator new(unsigned int)`. Your custom function should really match that (`static void* OperatorNew(size_t)` in the header declaration), and also return the result of the 0x577360 call.

Wanted9811 replied:
Yeah, thanks for letting me notice this!

0x1F9F1 commented on src/handlers/feature_handlers.cpp:
```diff
@@ -6645,15 +6645,30 @@ void vehSirenHandler::Install() {
-    //jmp out ltLightPool destructor
-    InstallPatch({ 0xEB }, {
-        0x4D6638,
+    //set old size of vehSiren to 0
```
This patch doesn't really do anything, since you are hooking the subsequent `operator new` call, and ignoring the parameter it corresponds to.

Wanted9811 committed:
Remove the latest patch

This patch does nothing since we are hooking the subsequent operator new call, and ignoring the parameter it corresponds to.

Wanted9811 committed:
Make OperatorNew return vehSiren* type

0x1F9F1 commented on src/handlers/feature_handlers.h:
```diff
@@ -421,7 +421,7 @@ class vehSirenHandler {
     void Update();
     void Reset();
-    void SizeOf();
+    MM2::vehSiren* OperatorNew();
```
You also need to fix calling convention (by making it static) and adding the size parameter

Wanted9811 replied:
Roger that

Wanted9811 committed:
Make OperatorNew static and add the size parameter to it
